src/semaphore.c

```diff
@@ -716,6 +771,11 @@ _dispatch_thread_semaphore_create(void)
 	int ret = sem_init(s4, 0, 0);
 	DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 	return (_dispatch_thread_semaphore_t)s4;
+#elif USE_FUTEX_SEM
+	dispatch_futex_t *futex = malloc(sizeof(*futex));
```

Comment:
you don't need that once the TSD patch is in, there should be an inlined `struct dispatch_futex_s` in that struct that is used as the "thread semaphore" to avoid a malloc for 64bits.